test/FunctionalTestUtils/TelemetryTestsBase.cs

```diff
@@ -107,14 +110,20 @@ public void ValidateBasicDependency(string assemblyName, string requestPath, Fun
             IEnumerable<DependencyTelemetry> dependencies = server.BackChannel.Buffer.OfType<DependencyTelemetry>();
             Assert.NotNull(dependencies);
             Assert.NotEmpty(dependencies);
-            Assert.Contains(dependencies,
-                d => d.Name == expected.Name
+
+            var dependencyTelemetry = dependencies.FirstOrDefault(d => d.Name == expected.Name
                   && d.Data == expected.Data
                   && d.Success == expected.Success
-                  && d.ResultCode == expected.ResultCode
-                );
+                  && d.ResultCode == expected.ResultCode);
+            Assert.NotNull(dependencyTelemetry);
+
+#if !NET451
```

Comment:
This is to test correlation, which doesn't work with NET451. It will work with NET46 though, and the code that make it work lives in the DiagnosticSource but only targeting NET46.